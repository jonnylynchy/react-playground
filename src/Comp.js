import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Comp extends Component {
	static propTypes = {
		render: PropTypes.func.isRequired,
	};

	state = {
		message: "You haven't clicked yet..."
	};

	handleClick = _ => {
		this.setState({
			message: "Clickity click click yo!"
		});
	};

	render() {
		return (
			<div style={{ height: '100%' }} onClick={this.handleClick}>
				{this.props.render(this.state)}
			</div>
		);
	}
}

export default Comp;