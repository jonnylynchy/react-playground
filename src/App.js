import React, { Component } from 'react';
import Comp from './Comp';

import './App.css';

class App extends Component {
	render() {
		return (
			<div style={{ height: '100%' }}>
				<Comp
					render={({ message }) => (
					// The render prop gives us the state we need
					// to render whatever we want here.
					<h1>
						{ message }
					</h1>
					)}
				/>
			</div>
		);
	}
};

export default App;
