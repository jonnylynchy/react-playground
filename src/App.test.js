import React from 'react';
import { shallow, render, mount, configure, renderProp } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './App';
import Comp from './Comp';

configure({ adapter: new Adapter() });

describe('when <App /> renders', () => {
	// One common solution is to instantiate another wrapper for the inner component:
	it('should render <Comp /> with a passed message using .prop()', () => {
		const wrapper = shallow(<App />);
		const compWrapper = shallow(
			wrapper.find(Comp).prop('render')({ message: "Wow, that was easy." })
		);

		console.log('Using .prop("render"):\n', compWrapper.html())

		// works
		expect(compWrapper.text()).toEqual("Wow, that was easy.");

		// not ideal
		expect(compWrapper.html()).toMatchSnapshot();
	});

	it('should render <Comp /> with a passed message using .renderProp()', () => {
		const betterWrapper = shallow(<App />)
			.find(Comp)
			.renderProp('render')({ message: "foo foo magoo" });
		
		console.log('Using .renderProp():\n', betterWrapper.html());

		expect(betterWrapper.html()).toMatchSnapshot();
	});
});
