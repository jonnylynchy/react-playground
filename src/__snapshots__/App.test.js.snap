// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when <App /> renders should render <Comp /> with a passed message using .prop() 1`] = `"<h1>Wow, that was easy.</h1>"`;

exports[`when <App /> renders should render <Comp /> with a passed message using .renderProp() 1`] = `"<h1>foo foo magoo</h1>"`;
